implementation/domain/web/jbb-board-web/src/main/java/org/jbb/board/web/forum/controller/AcpForumCategoryController.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (C) 2017 the original author or authors.
+ *
+ * This file is part of jBB Application Project.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  You may obtain a copy of the License at
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package org.jbb.board.web.forum.controller;
+
+import com.google.common.collect.Lists;
+
+import org.jbb.board.api.exception.ForumCategoryException;
+import org.jbb.board.api.model.ForumCategory;
+import org.jbb.board.api.service.BoardService;
+import org.jbb.board.api.service.ForumCategoryService;
+import org.jbb.board.web.forum.data.ForumCategoryRow;
+import org.jbb.board.web.forum.form.ForumCategoryDeleteForm;
+import org.jbb.board.web.forum.form.ForumCategoryForm;
+import org.jbb.lib.mvc.SimpleErrorsBindingMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Controller
+@RequestMapping("/acp/general/forums/category")
+public class AcpForumCategoryController {
+    private static final String VIEW_NAME = "acp/general/forumcategory";
+    private static final String REDIRECT_TO_FORUM_MANAGEMENT = "redirect:/acp/general/forums";
+
+    private static final String CATEGORY_FORM = "forumCategoryForm";
+    private static final String CATEGORY_DELETE_FORM = "forumCategoryDeleteForm";
+    private static final String CATEGORY_ROW = "forumCategory";
+
+    private final BoardService boardService;
+    private final ForumCategoryService forumCategoryService;
+    private final SimpleErrorsBindingMapper errorMapper;
+
+    @Autowired
+    public AcpForumCategoryController(BoardService boardService,
+                                      ForumCategoryService forumCategoryService,
+                                      SimpleErrorsBindingMapper errorMapper) {
+        this.boardService = boardService;
+        this.forumCategoryService = forumCategoryService;
+        this.errorMapper = errorMapper;
+    }
+
+    @RequestMapping(method = RequestMethod.GET)
+    public String forumCategoryGet(@RequestParam(value = "id", required = false) Long categoryId, Model model) {
```

Comment:
Why categoryID is not required parameter? 
Is invoke that method without these parameter has any motivation or point?